src/runners/rfq_maker_balance_cache_runner.ts

```diff
@@ -0,0 +1,222 @@
+    const start = new Date().getTime();
+
+    if (!MAKER_TOKEN_CACHE) {
+        MAKER_TOKEN_CACHE = createResultCache<any[]>(
+            () =>
+                connection
+                    .getRepository(MakerBalanceChainCacheEntity)
+                    .createQueryBuilder('maker_balance_chain_cache')
+                    .select(['maker_balance_chain_cache.tokenAddress', 'maker_balance_chain_cache.makerAddress'])
+                    .getMany(),
+            CACHE_MAKER_TOKENS_FOR_MS,
+        );
+    }
+    const results = (await MAKER_TOKEN_CACHE.getResultAsync()).result;
+
+    MAKER_BALANCE_CACHE_RESULT_COUNT.set(results.length);
+    MAKER_BALANCE_CACHE_RETRIEVAL_TIME.set(new Date().getTime() - start);
+
+    return results;
+}
+
+function splitValues(makerTokens: MakerBalanceChainCacheEntity[]): BalancesCallInput {
+    const functionInputs: BalancesCallInput = { addresses: [], tokens: [] };
+
+    return makerTokens.reduce(({ addresses, tokens }, makerToken) => {
+        return {
+            addresses: addresses.concat(makerToken.makerAddress!),
+            tokens: tokens.concat(makerToken.tokenAddress!),
+        };
+    }, functionInputs);
+}
+
+/**
+ * Returns the balaceChecker interface given a random address
+ */
+export function getBalanceCheckerContractInterface(
+    contractAddress: string,
+    provider: SupportedProvider,
+): BalanceCheckerContract {
+    return new BalanceCheckerContract(contractAddress, provider, { gas: BALANCE_CHECKER_GAS_LIMIT });
+}
+
+async function getErc20BalancesAsync(
+    balanceCheckerContractInterface: BalanceCheckerContract,
+    balancesCallInput: BalancesCallInput,
+    // HACK: allow for testing on ganache without override
+    codeOverride: boolean,
+): Promise<string[]> {
+    // due to gas contraints limit the call to 1K balance checks
+    const addressesChunkedArray = _.chunk(balancesCallInput.addresses, MAX_BALANCE_CHECKS_PER_CALL);
+    const tokensChunkedArray = _.chunk(balancesCallInput.tokens, MAX_BALANCE_CHECKS_PER_CALL);
+
+    const balanceCheckerByteCode = _.get(artifacts.BalanceChecker, 'compilerOutput.evm.deployedBytecode.object');
+
+    const balances = await Promise.all(
+        _.zip(addressesChunkedArray, tokensChunkedArray).map(async ([addressesChunk, tokensChunk]) => {
+            return !codeOverride
+                ? balanceCheckerContractInterface.balances(addressesChunk!, tokensChunk!).callAsync()
+                : balanceCheckerContractInterface.balances(addressesChunk!, tokensChunk!).callAsync(
+                      {
+                          overrides: {
+                              [RANDOM_ADDRESS]: {
+                                  code: balanceCheckerByteCode,
+                              },
+                          },
+                      },
+                      BlockParamLiteral.Latest,
+                  );
+        }),
+    );
+
+    const balancesFlattened = Array.prototype.concat.apply([], balances);
```

Comment:
Alternative implementation using [flatten](https://lodash.com/docs/4.17.15#flatten)

```typescript
const balancesFlattened = _.flatten(balances)
```

Reply:
I'll keep it as is for now